pkg/secure/mqttfactory_test.go

```diff
@@ -143,7 +143,6 @@ func TestConfigureMQTTClientForAuthWithCACert(t *testing.T) {
 
 	require.NoError(t, err)
 	assert.NotNil(t, target.opts.TLSConfig.RootCAs)
-	//assert.Empty(t, target.opts.Username) // Username may be required when cert authentication
 	assert.Empty(t, target.opts.Password)
```

Comment:
Same for all places you removed the assert.Empty.

```suggestion
        assert.Equal(t, target.opts.Username, "Username") 
	assert.Empty(t, target.opts.Password)
```